src/corppa/ocr/collate_txt.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+"""
+Script to turn directories with multiple text files into a single JSON
+file containing text contents of all files with page numbers based
+on text filenames. (Page number logic is currently Gale-specific).
+"""
+
+import argparse
+import json
+import pathlib
+
+from tqdm import tqdm
+
+from corppa.utils.path_utils import find_relative_paths
+
+
+def page_number(filename: pathlib.Path) -> str:
+    # NOTE: this logic is currently specific to Gale/ECCO files,
+    # which look like CW0112029406_00180.txt
+
+    # split the file base/stem name by _ and take the last part
+    pagenum = filename.stem.split("_")[-1]
+    # return the number as a string; strip extra trailing zero
+    return pagenum[:-1]  # strip trailing zero
+
+
+def collate_txt(
+    input_dir: pathlib.Path, output_dir: pathlib.Path, show_progress: bool = True
+):
+    directories = 0
+    txt_files = 0
+    skipped = 0
+
+    # stack tqdm bars so we can briefly show status
+    status = tqdm(
+        desc="Collating",
+        bar_format="{desc}{postfix}",
+        disable=not show_progress,
+    )
+
+    for ocr_dir, files in tqdm(
+        find_relative_paths(input_dir, [".txt"], group_by_dir=True),
+        desc="Directories with text files",
+        disable=not show_progress,
+    ):
+        # output will be a json file based on name of the directory containing text files,
+        # with parallel directory structure to the source
+        output_file = (output_dir / ocr_dir.parent / ocr_dir.name).with_suffix(".json")
```

Comment:
This was confusing for parsing what is the directory vs. filename. Here's an alternative:

```suggestion
        output_file = output_dir / ocr_dir.parent / f"{ocr_dir.name}.json"
```